naga/src/back/msl/writer.rs

```diff
@@ -344,7 +344,9 @@ impl TypedGlobalVariable<'_> {
         let (space, access, reference) = match var.space.to_msl_name() {
             Some(space) if self.reference => {
                 let access = if var.space.needs_access_qualifier()
-                    && !self.usage.contains(valid::GlobalUse::WRITE)
+                    && !self
+                        .usage
+                        .intersects(valid::GlobalUse::WRITE | valid::GlobalUse::ATOMIC)
```

Comment:
Preface: Inexperienced in naga

Wasn't the discussion before that all Atomic Textures would be write anyway, so this doesn't actually need to change? There's no harm I guess though.